snowplow-tracker/src/main/java/com/snowplowanalytics/snowplow/tracker/events/AbstractEvent.java

```diff
@@ -107,25 +131,47 @@ public T trueTimestamp(long trueTimestamp) {
         }
     }
 
+    private static class Builder2 extends Builder<Builder2> {
+        @Override
+        protected Builder2 self() {
+            return this;
+        }
+    }
+
+    protected AbstractEvent() {
+        this(new Builder2());
+    }
+
     AbstractEvent(Builder<?> builder) {
 
         // Precondition checks
-        Preconditions.checkNotNull(builder.context);
+        Preconditions.checkNotNull(builder.customContexts);
         Preconditions.checkNotNull(builder.eventId);
         Preconditions.checkArgument(!builder.eventId.isEmpty(), "eventId cannot be empty");
 
-        this.context = builder.context;
+        this.customContexts = builder.customContexts;
         this.deviceCreatedTimestamp = builder.deviceCreatedTimestamp;
         this.trueTimestamp = builder.trueTimestamp;
         this.eventId = builder.eventId;
     }
 
+    /**
+     * @deprecated As of release 1.5.0, replaced by {@link #getContexts()}
+     *
+     * @return the events custom context
+     */
+    @Override
+    @Deprecated
+    public @NonNull List<SelfDescribingJson> getContext() {
+        return getContexts();
+    }
+
     /**
      * @return the events custom context
      */
     @Override
-    public List<SelfDescribingJson> getContext() {
-        return new ArrayList<>(this.context);
+    public @NonNull List<SelfDescribingJson> getContexts() {
```

Comment:
In this specific case `List<SelfDescribingJson> getContexts()` what I read is: _the method returns a list of contexts_. I think is common convention in Java world indicate arrays with plural rather than singular. So if we consider the list a _singular_ context we should rewrite the method in these ways:
- `EventContext getContext()` (where EventContext is a collection of entities)
OR
- `List<SelfDescribingJson> getEntities()` or `getContextList()` or `getContextEntities()`

Personally I prefer to just consider each entity a context and treat it as plural element, it's less disruptive in our code. Otherwise it makes sense to introduce Entity as a first class object as we often talk of "entities" when we refer to pieces of "context".

Reply:
I like the idea of introducing Entity. It's out of scope for this PR but perhaps makes sense for a MAJOR bump across the tracker estate.
Happy to leave this as it is at the moment.